integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -547,37 +554,43 @@ private int truncateInternal(String path, long size) {
     if (res != 0) {
       return res;
     }
-    FuseFileEntry<FuseFileStream> entry = mFileEntries.getFirstByField(PATH_INDEX, path);
-    if (entry != null) {
-      entry.getFileStream().truncate(size);
-      return 0;
-    }
-    Optional<URIStatus> status = AlluxioFuseUtils.getPathStatus(mFileSystem, uri);
-    if (!status.isPresent()) {
-      if (size == 0) {
+    try {
+      FuseFileEntry<FuseFileStream> entry = mFileEntries.getFirstByField(PATH_INDEX, path);
+      if (entry != null) {
+        entry.getFileStream().truncate(size);
         return 0;
       }
-      LOG.error("Failed to truncate file {} to {} bytes: file does not exist", path, size);
-      return -ErrorCodes.EEXIST();
-    }
-
-    if (status.get().isCompleted()) {
-      long fileLen = status.get().getLength();
-      if (fileLen == size) {
-        return 0;
+      Optional<URIStatus> status = AlluxioFuseUtils.getPathStatus(mFileSystem, uri);
+      if (!status.isPresent()) {
+        if (size == 0) {
+          return 0;
+        }
+        LOG.error("Failed to truncate file {} to {} bytes: file does not exist", path, size);
+        return -ErrorCodes.EEXIST();
       }
-      if (size == 0) {
-        AlluxioFuseUtils.deletePath(mFileSystem, uri);
+
+      if (status.get().isCompleted()) {
+        long fileLen = status.get().getLength();
+        if (fileLen == size) {
+          return 0;
+        }
+        if (size == 0) {
+          AlluxioFuseUtils.deletePath(mFileSystem, uri);
+        }
+        LOG.error("Failed to truncate file {}({} bytes) to {} bytes: not supported.",
+            path, fileLen, size);
+        return -ErrorCodes.EOPNOTSUPP();
       }
-      LOG.error("Failed to truncate file {}({} bytes) to {} bytes: not supported.",
-          path, fileLen, size);
+
+      LOG.error("Failed to truncate file {} to {} bytes: "
+              + "file is being written by other Fuse applications or Alluxio APIs.",
+          path, size);
+      return -ErrorCodes.EOPNOTSUPP();
+    } catch (UnimplementedRuntimeException e) {
```

Comment:
Will have future PRs transfer Alluxio related exception to FUSE related exception